src/Microsoft.VisualStudio.Threading.Tests/AsyncReaderWriterLockTests.cs

```diff
@@ -1813,6 +1813,49 @@ public void WriteLockAsyncSynchronousReleaseAllowsOtherWriters()
             Assert.True(firstLockTask.Wait(TestTimeout)); // rethrow any exceptions
         }
 
+        [StaFact]
+        public void WriteLockDisposingShouldNotBlockByOtherWriters()
+        {
+            var firstLockToRelease = new AsyncManualResetEvent();
+            var firstLockAccquired = new AsyncManualResetEvent();
+            var firstLockToDispose = new AsyncManualResetEvent();
+            var firstLockTask = Task.Run(async delegate
+            {
+                using (var firstLock = await this.asyncLock.WriteLockAsync())
+                {
+                    firstLockAccquired.Set();
+                    await firstLockToRelease.WaitAsync();
+                    await firstLock.ReleaseAsync();
+                    await firstLockToDispose.WaitAsync();
+                }
+            });
+
+            var secondLockReleased = new AsyncManualResetEvent();
+
+            var secondLockTask = Task.Run(async delegate
+            {
+                await firstLockAccquired.WaitAsync();
+                var awaiter = this.asyncLock.WriteLockAsync().GetAwaiter();
+                Assert.False(awaiter.IsCompleted);
+                awaiter.OnCompleted(() =>
```

Comment:
You should probably wrap this entire delegate in a `try/catch` that captures any exception so it can be rethrown by the test method so it forces a failure if it happens.

Reply:
in that case, it will not set secondLockReleased, and the test will fail.  But I updated the test case to make it more clear on this.